python/res/job_queue/job_queue_node.py

```diff
@@ -39,7 +39,10 @@ def __init__(self,job_script, job_name, run_path, num_cpu,
         argc = 1
         argv = StringList()
         argv.append(run_path)
-        self.started = False
+
+        self.thread_status = ThreadStatusType.THREAD_READY
+        self.thread = None
```

Comment:
Could we make the `thread` an implementation detail by extending the `JobQueueNode` interface with a `join` function?

Reply:
:+1: